File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-5.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
     int i=0;
     int j=nums.size()-1;
     int ans=nums[0];
        while(i<=j){
      if (nums[i]<nums[j]){
        return min(ans,nums[i]);
      }
      int m=(i+j)/2;
      ans=min(ans,nums[m]);
      if(nums[m]>=nums[i])
      {i=m+1;
      }else{ 
    
        j=m-1;

      }
     }
     return ans;
    }
};

File: Data Structures & Algorithms/merge-intervals/submission-5.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        i,n=0,len(intervals)
        if n==1:
            return intervals
        ans=[intervals[0]]    
        for a , b in intervals:
            y=ans[-1][1]
            if a<=y:
                ans[-1][1]=max(y,b)
            else:
                 ans.append([a,b])
        return ans